### ISSUE TITLE
syslog test fails without sockets

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2221]
Migrated from [rt.perl.org#3532](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3532) (status was 'resolved')

Searchable as RT3532$

**Comment by p5pRT** [comment]
### From @ysth

#### Created by @ysth



This is perl patched up to \#6388\.  \(Did not occur in vanilla 5\.6\.0\)

I get a test failure&#8203;:

lib/syslog\.\.\.\.\.\.\.\.\.\.\.Can't locate Socket\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./lib d&#8203;:/perl/lib/5\.6\.0/os2 d&#8203;:/perl/lib/5\.6\.0 d&#8203;:/perl/lib/site\_perl/5\.6\.0/os2 d&#8203;:/perl/lib/site\_perl/5\.6\.0 d&#8203;:/perl/lib/site\_perl \.\) at \.\./lib/Sys/Syslog\.pm line 12\.
BEGIN failed\-\-compilation aborted at \.\./lib/Sys/Syslog\.pm line 12\.
Compilation failed in require at lib/syslog\.t line 13\.
BEGIN failed\-\-compilation aborted at lib/syslog\.t line 13\.
Scalars leaked&#8203;: 1
FAILED at test 0

This is obviously because Socket wasn't built because d\_socket is undef
\(as it should be\)\.  Here's one way to fix it&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- Configure.orig	Thu Jul 13 18:03:14 2000
+++ Configure	Sat Jul 15 22:32:26 2000
@@ -14994,6 +14994,11 @@
 		true|$define|y) avail_ext="$avail_ext $xxx" ;;
 		esac
 		;;
+	Sys/Syslog|sys/syslog)
+		case "$d_socket" in 
+		true|$define|y) avail_ext="$avail_ext $xxx" ;;
+		esac
+		;;
 	Thread|thread)
 		case "$usethreads" in 
 		true|$define|y) avail_ext="$avail_ext $xxx" ;;

```
</details>
End of Patch.

This would still fail if d\_socket was defined but the Configure was
told not to do Socket\.pm\.  Is there a better way to handle module
dependencies in Configure?

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

This perlbug was built using Perl v5.6.0 - Thu Jul 13 18:05:05 PDT 2000
It is being executed now by  Perl v5.6.0 - Tue Jun 20 09:44:05 PDT 2000.

Site configuration information for perl v5.6.0:

Configured by sthoenna at Tue Jun 20 09:44:05 PDT 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=os2, osvers=2.30, archname=os2
    uname='os2 efn.org 2 2.30 i386  '
    config_args='-de -Dprefix=d:/perl -Aoptimize=-DDEBUGGING'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='gcc', optimize='-O2 -fomit-frame-pointer -malign-loops=2 -malign-jumps=2 -malign-functions=2 -s -DDEBUGGING', gccversion=2.8.1
    cppflags='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=62'
    ccflags ='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=62'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Zexe -Zomf -Zmt -Zcrtdll -Zstack 32000'
    libpth=d:/emx/lib d:/emx/lib/mt
    libs=-lsocket -lm -lbsd
    libc=d:/emx/lib/mt/c_import.lib, so=dll, useshrplib=true, libperl=libperl.lib
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-Zdll', lddlflags='-Zdll -Zomf -Zmt -Zcrtdll -s'

Locally applied patches:
    


@INC for perl v5.6.0:
    d:/perl/lib/5.6.0/os2
    d:/perl/lib/5.6.0
    d:/perl/lib/site_perl/5.6.0/os2
    d:/perl/lib/site_perl/5.6.0
    d:/perl/lib/site_perl
    .


Environment for perl v5.6.0:
    HOME=d:\home\sthoenna
    LANG=en_us
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=d:\bin;C:\OS2;d:\perl\bin;C:\OS2\SYSTEM;C:\OS2\INSTALL;C:\;C:\OS2\MDOS;C:\OS2\APPS;C:\MMOS2;d:\os2apps\util;d:\DOSAPPS\UTIL;c:\sio;D:\WINDOWS;d:\pdksh;d:\emx\bin;d:\emacs\19.33\bin;d:\ispell
    PERL_BADLANG (unset)
    PERL_SH_DIR=d:\BIN
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

> lib/syslog\.\.\.\.\.\.\.\.\.\.\.Can't locate Socket\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./lib d&#8203;:/perl/lib/5\.6\.0/os2 d&#8203;:/perl/lib/5\.6\.0 d&#8203;:/perl/lib/site\_perl/5\.6\.0/os2 d&#8203;:/perl/lib/site\_perl/5\.6\.0 d&#8203;:/perl/lib/site\_perl \.\) at \.\./lib/Sys/Syslog\.pm line 12\.
> BEGIN failed\-\-compilation aborted at \.\./lib/Sys/Syslog\.pm line 12\.
> Compilation failed in require at lib/syslog\.t line 13\.
> BEGIN failed\-\-compilation aborted at lib/syslog\.t line 13\.
> Scalars leaked&#8203;: 1
> FAILED at test 0
> 
> This is obviously because Socket wasn't built because d\_socket is undef
> \(as it should be\)\.  Here's one way to fix it&#8203;:

Thanks\.

> \-\-\- Configure\.orig    Thu Jul 13 18&#8203;:03&#8203;:14 2000
> \+\+\+ Configure    Sat Jul 15 22&#8203;:32&#8203;:26 2000
> @&#8203;@&#8203; \-14994\,6 \+14994\,11 @&#8203;@&#8203;
>          true|$define|y\) avail\_ext="$avail\_ext $xxx" ;;
>          esac
>          ;;
> \+    Sys/Syslog|sys/syslog\)
> \+        case "$d\_socket" in 
> \+        true|$define|y\) avail\_ext="$avail\_ext $xxx" ;;
> \+        esac
> \+        ;;
>      Thread|thread\)
>          case "$usethreads" in 
>          true|$define|y\) avail\_ext="$avail\_ext $xxx" ;;
> End of Patch\.
> 
> This would still fail if d\_socket was defined but the Configure was
> told not to do Socket\.pm\.  Is there a better way to handle module
> dependencies in Configure?

Unfortunately\, no\.
